modules/PatternUtils.js

```diff
@@ -149,7 +149,7 @@ export function formatPattern(pattern, params) {
   params = params || {}
 
   const { tokens } = compilePattern(pattern)
-  let parenCount = 0, pathname = '', splatIndex = 0
+  let parenCount = 0, pathname = '', splatIndex = 0, history = []
```

Comment:
I really wouldn't use a variable named `history`, given that this library uses that name extensively in various places.